packages/core/src/storage/compare.ts

```diff
@@ -70,7 +106,64 @@ export class StorageLayoutComparator {
   constructor(readonly unsafeAllowCustomTypes = false, readonly unsafeAllowRenames = false) {}
 
   compareLayouts(original: StorageItem[], updated: StorageItem[]): LayoutCompatibilityReport {
-    return new LayoutCompatibilityReport(this.layoutLevenshtein(original, updated, { allowAppend: true }));
+    if (original.every(isFullInfo) && updated.every(isFullInfo)) {
+      // We are going to cut the layout in sections following the original gaps.
+      let ptr = 0;
+      const sections: { begin: number; end: number }[] = [];
+
+      // For each gap in the original layout, we try to do a cut. If that is possible, we isolate
+      // the section that is before the cut and continue looking for other sections.
+      const gaps = original.filter(entry => entry.label === '__gap');
+      for (let i = 0; i < gaps.length; ++i) {
+        // Beginning of the gap(s) is the end of the previous section
+        const gapBegin = storageItemBegin(gaps[i]);
+
+        // Merge consecutive gaps
+        for (; i < gaps.length - 1; ++i) {
+          if (storageItemEnd(gaps[i]) !== storageItemBegin(gaps[i + 1])) {
+            break;
+          }
+        }
+
+        // End of the gap(s) is the begining of the next section
+        const gapEnd = storageItemEnd(gaps[i]);
```

Comment:
I think we are making assumptions that the gap has the right format. If someone was to put `bool private __gap` in their system, it would possibly break things.

Maybe we need to check in the filter (line 116) that the gap is a struct (that enforces begin and end are aligned with slots)

Reply:
Should non-array gaps be considered a gap?
e.g. `bool private __gap` or `uint256 private __gap`

Reply:
Added a check to ensure the gap is an array.  Is struct relevant?

Reply:
I think this is good !